src/sage/combinat/partition.py

```diff
@@ -2485,6 +2485,118 @@ def conjugate(self):
         par = Partitions_n(sum(self))
         return par.element_class(par, conjugate(self))
 
+    def glaisher_franklin(self, s):
+        r"""
+        Apply the Glaisher-Franklin bijection to ``self``.
+
+        The Franklin-Glaisher bijection, with parameter `s`, returns
+        a partition whose set of parts that are repeated at least `s`
+        times equals the set of parts divisible by `s` in ``self``,
+        after dividing each part by `s`.
+
+        INPUT:
+
+        - ``s`` -- positive integer
+
+        EXAMPLES::
+
+            sage: Partition([4, 3, 2, 2, 1]).glaisher_franklin(2)
+            [3, 2, 2, 1, 1, 1, 1, 1]
+
+        TESTS:
+
+        The map preserves the size::
+
+            sage: all(mu.glaisher_franklin(s).size() == n
+            ....:     for n in range(20) for mu in Partitions(n)
+            ....:     for s in range(1, 5))
+            True
+
+        The map is bijective::
+
+            sage: l = [[mu.glaisher_franklin(s)
+            ....:      for n in range(20) for mu in Partitions(n)]
+            ....:     for s in range(1, 5)]
+            sage: all(len(set(ls)) == len(ls) for ls in l)
+            True
+
+        The map transports the statistics::
+
+            sage: d = lambda la, s: set(p / s for p in la if p % s == 0)
+            sage: r = lambda la, s: set(p for p in la if list(la).count(p) >= s)
+            sage: all(d(mu, s) == r(mu.glaisher_franklin(s), s)
+            ....:     for n in range(20) for mu in Partitions(n)
+            ....:     for s in range(1, 5))
+            True
+
+        For `s=2`, the map is known to findstat::
+
+            sage: findmap(Partitions, lambda mu: mu.glaisher_franklin(2))       # optional - internet
+            0: Mp00312 (quality [100])
+        """
+        s = ZZ(s)
+        if s.is_one():
+            return self
+        mu = []
+        for p, m in enumerate(self.to_exp(), 1):
+            if not p % s:
+                mu.extend([p // s] * (m*s))
+            else:
+                mu.extend([p1]*v for i, v in enumerate(m.digits(s))
+                          if (p1 := p * s**i))
```

Comment:
```suggestion
                mu.extend([p1]*v for i, v in enumerate(m.digits(s))
                          if (p1 := p * s**i))
```
This doesn't quite work here. Sorry for that. How you had it before is necessary as it extends by adding the lists over the `for` loop. Also, isn't `p1` always nonzero since $p, s \in \mathbb{Z}_{>0}$?

Reply:
Yes, `p1` is nonzero, the `if` is only to define it, because it is the same for all `v`, and previously I was looping over `v`.

Reply:
I guess
```
                for i, v in enumerate(m.digits(s)):
                    mu.extend([p * s**i]*v)
```
is better, right?

Reply:
Possibly. A quick timing check should be done. Although it is a mirco-optimization at the end of the day, so I might side with the more readable one here.

For `p1`, what I am saying is it will *never* be nonzero, no matter what `p` and `s` are as it is a product of positive numbers.

Reply:
Yes, 'p1' is nonzero, why are you asking?

Reply:
But by its formula, it is always nonzero, so the `if` check is vacuous here.

Reply:
The if check was only to be able to define 'p1'.

Reply:
I see. Then yes, the `for` loop with the `mu.extend` is better.